pkg/k8s/k8s_test.go

```diff
@@ -369,6 +368,51 @@ func Test_NewDeployment(t *testing.T) {
 				return d
 			},
 		},
+		{
```

Comment:
I think it will work correctly, but I think it would be nice to have a test case setting `FilesRef.Files` to nil. It should leave `ConfigMapVolumeSource.Items` as nil and mount all files at the root level.

Reply:
Sure. 👍